framework/src/utils/GriddedData.C

```diff
@@ -57,6 +57,77 @@ GriddedData::GriddedData(std::string file_name)
   parse(_dim, _axes, _grid, _fcn, _step, file_name);
 }
 
+GriddedData::GriddedData(std::vector<Real> x_axis,
+                         std::vector<Real> y_axis,
+                         std::vector<Real> z_axis,
+                         std::vector<Real> t_axis,
+                         std::vector<Real> fcn)
+{
+  unsigned int num_data_pt = 1;
+  std::vector<std::vector<Real>> grid;
+  unsigned int dim = 0;
+  _axes.resize(0);
+  // check the sizes of each axis, if one is zero, do not count it in _dim.
+  if (x_axis.size() != 0)
+  {
+    dim += 1;
+    _axes.push_back(0);
+    grid.resize(dim); // add another dimension to the grid
+    grid[dim - 1].resize(0);
+    grid[dim - 1] = x_axis;
+    num_data_pt *= grid[dim - 1].size();
+  }
+  if (y_axis.size() != 0)
+  {
+    dim += 1;
+    _axes.push_back(1);
+    grid.resize(dim); // add another dimension to the grid
+    grid[dim - 1].resize(0);
+    grid[dim - 1] = y_axis;
+    num_data_pt *= grid[dim - 1].size();
+  }
+  if (z_axis.size() != 0)
```

Comment:
put the contents of this if statement into a function that will fill the _grid, _axes and _dim member data.  The return value can be num_data_pt for the checking at the end.  Or you can compute num_data_pt at the end from the grid data

Reply:
Done.